src/crypto-api/keybackup.ts

```diff
@@ -40,3 +43,42 @@ export interface KeyBackupInfo {
     etag?: string;
     version?: string; // number contained within
 }
+
+export interface KeyBackupStatus {
+    version: string;
+    enabled: boolean;
+}
+
+export type SigInfo = {
+    deviceId: string;
+    valid?: boolean | null; // true: valid, false: invalid, null: cannot attempt validation
+    device?: DeviceInfo | null;
+    crossSigningId?: boolean;
+    deviceTrust?: DeviceTrustLevel;
+};
+
+export type TrustInfo = {
+    usable: boolean; // is the backup trusted, true iff there is a sig that is valid & from a trusted device
+    sigs: SigInfo[];
+    // eslint-disable-next-line camelcase
+    trusted_locally?: boolean;
+};
+
+export interface IKeyBackupCheck {
+    backupInfo?: IKeyBackupInfo;
+    trustInfo: TrustInfo;
+}
+
+export interface SecureKeyBackup {
+    getKeyBackupStatus(): Promise<KeyBackupStatus | null>;
+
+    stop(): void;
+
+    /**
+     * Check the server for an active key backup and
+     * if one is present and has a valid signature from
+     * one of the user's verified devices, start backing up
+     * to it.
+     */
+    checkAndStart(): Promise<IKeyBackupCheck | null>;
+}
```

Comment:
these things all need good doc-comments explaining the concepts they represent and what each of the properties are for.

Reply:
ok, will try revert engineer it ;)